sklearn/base.py

```diff
@@ -435,6 +437,22 @@ def _validate_data(self, X, y=None, reset=True,
 
         return out
 
+    def _repr_html_(self):
+        """HTML or string representation of an estimator depending on
+        global configuration flag `repr_html`.
+
+        Read more in the :ref:`User Guide <visualizing_composite_estimators>`.
+
+        Returns
+        -------
+        repr: str
+            HTML or string representation of estimator.
+        """
+        repr_html = get_config()["repr_html"]
+        if repr_html:
+            return estimator_repr_html(self)
+        return f"<code>{repr(self)}</code>"
```

Comment:
```suggestion
        return None
```
at least in the version of Jupyter I tried, this is better. @amuller is correct.

Reply:
As per earlier comment, overriding `_repr_html_`  and relying on the returning of `None` is brittle.

I recommend only implementing `_repr_mimebundle_`  and populating it with `get_config()["repr_html"]`.

Reply:
I am trying to understand the `_repr_mimbundle_` formatter as documented here but I find it quite confusing:

https://ipython.readthedocs.io/en/stable/api/generated/IPython.core.formatters.html#IPython.core.formatters.MimeBundleFormatter

@SylvainCorlay do you a good example of code that uses `_repr_mimebundle_` correctly?

Reply:
Indeed, this is not great documentation. `_repr_mimebundle_` returns a dictionary of mime types, which is serialized as JSON

```Python
{
    'text/html': '<div class="foo">Hello Notebook</div>',
    'text/plain': 'Hello Console'
}
```

You can add arbitrary mimetypes as long as you can handle them in the front-end.

Reply:
![hub gke mybinder org_user_jupyter-widgets-ipywidgets-dnin8tek_notebooks_docs_source_examples_Untitled ipynb_kernel_name=python3(QuantStack)](https://user-images.githubusercontent.com/2397974/80384305-013d5700-88a5-11ea-811b-1b33eab0acff.png)

![Capture](https://user-images.githubusercontent.com/2397974/80384331-0b5f5580-88a5-11ea-98d0-4de3665e534f.PNG)

Reply:
`_repr_mimebundle_` is actually nicer because you can add arbitrary mimetypes.

For example, the https://github.com/jupyterlab/jupyter-renderers projects registers a bunch of mimetypes that anyone can target if the front-end is installed.

So I think a nice way to approach this rather than merely an "opt-in" option would be a plugin system with a configuration directory where third-parties can add new mime types for scikit-learn types.